core/pow/engine.go

```diff
@@ -157,6 +158,26 @@ func (e *Engine) EndOfBlock() {
 	e.lock.Lock()
 	defer e.lock.Unlock()
 
+	// run this once for migration cleanup
+	// this is going to clean up blocks that belong to inactive states which are unreachable by the latter loop.
+	if e.lastPruningBlock == 0 {
+		if e.activeParams[0].fromBlock > 0 {
+			end := e.activeParams[0].fromBlock
+			for block := uint64(0); block <= end; block++ {
+				if b, ok := e.heightToTx[block]; ok {
+					for _, v := range b {
+						delete(e.seenTx, v)
+					}
+				}
+				for _, v := range e.heightToTid[block] {
+					delete(e.seenTid, v)
+				}
+				delete(e.heightToTx, block)
+				delete(e.heightToTid, block)
+			}
+		}
+	}
+
 	toDelete := []int{}
 	// iterate over parameters set and clear then out if ther's not relevant anymore.
```

Comment:
```suggestion
	// iterate over parameters set and clear then out if they're not relevant anymore.
```

I realise this is out of scope of the PR but this typo hurts my eyes

Reply:
I hear you :)